### PR TITLE
Also show kernel modules like [uprobes].

### PR DESCRIPTION
**Pull request by florian-kuebler**
However, those modules can not be loaded.

The main reason for this PR is, to be able to detect, that a certain address is a uprobe.
However, some of those "named maps" also show up in samples, resulting in having "???" as module, rather than e.g. "[stack]". So the experience would be a bit nicer.

### CODE REVIEW

**Comment by ronaldfw** [comment]
What is the purpose of showing these modules? Can you add a "why" to the PR description? Is there a buganizer bug that you could link?

File: OrbitCore/LinuxUtils.cpp [hunk 1]
```diff
@@ -96,6 +96,7 @@ void ListModules(pid_t pid,
   struct AddressRange {
     uint64_t start_address;
     uint64_t end_address;
+    bool is_kernel_module;
```
Comment (ronaldfw) [inline OrbitCore/LinuxUtils.cpp:99]: This should not be a boolean, but rather an enum ModuleType (or something equivalent).

Comment (dpallotti) [inline OrbitCore/LinuxUtils.cpp:99]: Doesn't "kernel module" have a very specific meaning, quite distant from this one? It seems that a simple "is_loadable" would be more appropriate (under the hood this seems to be whether the map is mapped to a file).

File: OrbitCore/LinuxUtils.cpp [hunk 2]
```diff
@@ -107,7 +108,9 @@ void ListModules(pid_t pid,
 
     // tokens[4] is the inode column. If inode equals 0, then the memory is not
     // mapped to a file (might be heap, stack or something else)
-    if (tokens.size() != 6 || tokens[4] == "0") continue;
+    if (tokens.size() != 6) continue;
```
Comment (dpallotti) [inline OrbitCore/LinuxUtils.cpp:111]: So this excludes anonymous maps, right? Then the comment above refers not to this line but the next one (and this one should have its own comment).

File: OrbitCore/LinuxUtils.cpp (hunk 1 above)
Comment (florian-kuebler) [reply OrbitCore/LinuxUtils.cpp:99]: Right naming is probably not that nice. I was also thinking of something like "is_loadable". However, the "m_FoundPdb" dies something very similar.

Comment (dpallotti) [reply OrbitCore/LinuxUtils.cpp:99]: Either way, the name is incorrect, these maps are not necessarily related to the kernel, they are just maps with a name.

Comment (florian-kuebler) [reply OrbitCore/LinuxUtils.cpp:99]: Any suggestion? "is_loadable" vs. "can_be_loaded"

**Comment by ronaldfw** [comment]
As discussed, this should not have the UI changes, right? Are you going to close this one or change it as we agreed earlier?

**Pull request closed without merging**